src/utils/helpers.py

```diff
@@ -1,33 +1,73 @@
 from pathlib import Path
 
+import jwt
+import requests
 
-def load_predicates(path: str) -> list[str]:
+
+def create_jwt_token(gh_app_id: str, gh_pem_path: str) -> str:
     """
-    Fetch BETWEEN predicates from a provided SQL file.
+    Generates a JSON web token for authentication with the GitHub API.
 
     Args:
-        path: String path to a SQL file within the `config/`
-            directory. The SQL file should define SQL BETWEEN expressions,
-            where each expression is one chunk that will be extracted by
-            Spark during JDBC reads. Expressions should not be overlapping.
+        gh_app_id: The GitHub App ID.
+        gh_pem_path: The path to the GitHub App's private key file.
 
     Returns:
-        A list of SQL predicate strings used to divide a table into chunks
-        during JDBC reads.
+        str: The JWT token for authentication.
     """
-    full_path = "/tmp/config" / Path(path)
-    with open(full_path.resolve().as_posix(), mode="r") as file:
-        predicates = [line.strip() for line in file.readlines()]
 
-    return predicates
+    with open(gh_pem_path, "rb") as pem_file:
+        signing_key = jwt.jwk_from_pem(pem_file.read())
 
+    payload = {
+        "iat": int(time.time()),
+        "exp": int(time.time()) + 60,
+        "iss": gh_app_id,
+    }
 
-def strip_table_prefix(table_name: str) -> str:
-    """Removes the 'iasworld.' prefix from a table name."""
-    if table_name.startswith("iasworld."):
-        return table_name.split(".", 1)[1]
-    return table_name
+    jwt_instance = jwt.JWT()
+    encoded_jwt = jwt_instance.encode(payload, signing_key, alg="RS256")
+
+    return encoded_jwt
+
+
+def dispatch_gh_worfklow(gh_jwt: str, gh_repo: str, gh_workflow: str) -> None:
+    """
+    Dispatch a GitHub Actions workflow using the GitHub API.
+
+    Args:
+        gh_jwt: The JWT token for authentication with the GitHub API. Returned
+             by the `create_jwt_token` function.
+        gh_repo: API URL for the target repository containing a workflow.
+        gh_workflow: Workflow YAML file, relative to `.github/workflows`.
+    """
+
+    def create_headers(bearer: str) -> dict:
+        """Create headers with different bearers for the GitHub API requests"""
+        headers = {
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {bearer}",
+            "X-GitHub-Api-Version": "2022-11-28",
+        }
+        return headers
+
+    response = requests.get(
+        "https://api.github.com/app/installations",
+        headers=create_headers(gh_jwt),
+    )
+    gh_tokens_url = response.json()[0]["access_tokens_url"]
+
+    response = requests.post(gh_tokens_url, headers=create_headers(gh_jwt))
+    gh_token = response.json()["token"]
+
+    data = {"ref": "master"}
+    response = requests.post(
+        f"{gh_repo}/actions/workflows/{gh_workflow}/dispatches",
+        headers=create_headers(gh_token),
+        json=data,
+    )
```

Comment:
**[Suggestion, non-blocking]** It might behoove us to throw some [`raise_for_status()`](https://3.python-requests.org/user/quickstart/#response-status-codes) calls in here so that we can get ahead of any HTTP errors. I don't think we need to go too deep on logging just yet, but promptly raising HTTP errors before we run into KeyErrors or JSON formatting errors seems like low-hanging fruit.

Reply:
I'm about to handle logging and raising errors in the next PR, so going to put it off until then!